File: template/app/src/main/java/com/xmartlabs/template/module/RestServiceModule.java
package com.xmartlabs.template.module;

import android.content.Context;

import com.xmartlabs.template.R;
import com.xmartlabs.template.App;

import okhttp3.HttpUrl;

public class RestServiceModule extends com.xmartlabs.bigbang.retrofit.module.RestServiceModule {
  private static final String BASE_URL = App.context.getResources().getString(R.string.base_url);

  @Override
  public HttpUrl provideBaseUrl(Context context) {
    return HttpUrl.parse(BASE_URL);
  }
}

File: template/app/src/main/java/com/xmartlabs/template/module/RestServiceModule.kt
package com.xmartlabs.template.module

import android.content.Context
import com.xmartlabs.template.App
import com.xmartlabs.template.R
import okhttp3.HttpUrl
import com.xmartlabs.bigbang.retrofit.module.RestServiceModule as CoreRestServiceModule

class RestServiceModule : CoreRestServiceModule() {
  companion object {
    private val BASE_URL = App.context.resources.getString(R.string.base_url)
  }

  override fun provideBaseUrl(context: Context) = HttpUrl.parse(BASE_URL)
}
